### DISCUSSION TITLE
IsBlankOrWhitespace?

### DISCUSSION THREAD

**Discussion by daryllabar** [microsoft/Power-Fx#945] [Q&A] [answered]
It is very common to check for a text field to be populated using `IsBlank(TextInput1.Text)` but unfortunetly this will return false for `" "` or `"\t"' or even new lines.  What's the best way to handle this?  

I came up with `IsBlank(TextInput1.Text) || !IsMatch(TextInput1.Text, NonSpace, Contains & Multiline)` but @MikeStall's version makes a whole lot more sense: `IsBlank(Trim(TextInput1.Text))`

**Comment by MikeStall** [comment, marked as answer]
Yeah, the Trim() function is the key here. 
https://github.com/microsoft/Power-Fx/blob/main/src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Trim.txt

**Comment by daryllabar** [comment]
Apparently Chat GPT thinks `Len(Trim(TextInput1.Text)) = 0` is better
![image](https://user-images.githubusercontent.com/9728423/210282125-effdf05f-d49d-4bd1-a487-316e8d7d53e8.png)
